sonarqube/content.md

```diff
@@ -30,68 +45,145 @@ $ mvn sonar:sonar -Dsonar.host.url=http://$(boot2docker ip):9000
+```
+3. Configure sonar.properties to configure the database settings. Templates are available for every supported database. Just uncomment and configure the template you need and comment out the lines dedicated to H2:
+```
+Example for PostgreSQL
+sonar.jdbc.username=sonarqube
+sonar.jdbc.password=mypassword
+sonar.jdbc.url=jdbc:postgresql://localhost/sonarqube
+```
+
+4. Drivers for the supported databases (except Oracle) are already provided. Do not replace the provided drivers; they are the only ones supported.
 
-### Option 1: Use parameters via Docker environment variables
+For Oracle, copy the JDBC driver into $SQ_HOME/extensions/jdbc-driver/oracle.
 
-The production database is configured with the following SonarQube properties used as environment variables: `sonar.jdbc.username`, `sonar.jdbc.password` and `sonar.jdbc.url`.
+5. By default, SonarQube is configured to run on any computer with a simple Java JRE.
+
+For better performance, the first thing to do when installing a production instance is to use a Java JDK and activate the server mode by uncommenting/setting the following line in $SQ_HOME/conf/sonar.properties:
+```
+sonar.web.javaOpts=-server
+```
+To change the Java JVM used by SonarQube, simply edit $SQ_HOME/conf/wrapper.conf and update the following line:
+```
+wrapper.java.command=/path/to/my/jdk/bin/java
+```
 
+6. Run the image:
 ```console
 $ docker run -d --name sonarqube \
     -p 9000:9000 \
-    -e sonar.jdbc.username=sonar \
-    -e sonar.jdbc.password=sonar \
-    -e sonar.jdbc.url=jdbc:postgresql://localhost/sonar \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/logs:/opt/sonarqube/logs \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
     sonarqube
 ```
 
-Use of the environment variables `SONARQUBE_JDBC_USERNAME`, `SONARQUBE_JDBC_PASSWORD` and `SONARQUBE_JDBC_URL` is deprecated, and will stop working in future releases.
+## Upgrade SonarQube
+Follow these steps to upgrade SonarQube:
 
-More recipes can be found [here](https://github.com/SonarSource/docker-sonarqube/blob/master/recipes.md).
+1. Create a new `sonarqube_home` folder and backup your old folder.
 
-You can pass `sonar.` configuration properties as Docker environment variables, as demonstrated in the example above for database configuration.
+2. Update the var env `$SQ_HOME` pointing to it. Backup the old one:
+```console
+$ mkdir /path/to/your/filesystem/sq_home81
+$ export SQ_HOME=/path/to/your/filesystem/sq_home81
+```
+
+3. Initialize the new `SQ_HOME` with `--init`:
+```console
+$ docker run --rm \
+    -v $SQ_HOME/conf:/opt/sonarqube/conf \
+    -v $SQ_HOME/extensions:/opt/sonarqube/extensions \
+    -v $SQ_HOME/data:/opt/sonarqube/data \
+    sonarqube --init  
+```
+
+4. Follow the normal upgrade procedure https://docs.sonarqube.org/latest/setup/upgrading/
+  - Manually install the non-default plugins
+  - Update the contents of sonar.properties and wrapper.conf
+  - etc.  
```

Comment:
The "normal" upgrade procedure mentions steps that are not required here (download, unzip, restart...). I would not recommend to blindly follow it :)
I feel we should either repeat the relevant warning or steps (which can be fastidious), or slightly revisit what we say.
Maybe we could recommend to take a look at the Upgrade Guide and to, more specifically, 
 - Manually install the non-default plugins
 - Update the contents of sonar.properties and wrapper.conf
 - Copy the Oracle DB JDBC driver if needed